### ISSUE TITLE
unshiftContainer seems to incorrectly handle function params

### ISSUE THREAD

**Issue by achambers** [babel/babel#6150]
### Synopsis

`unshiftContainer` does not seem to perform correctly when unshifting `arguments` of a `CallExpression`.

### Input Code
https://astexplorer.net/#/gist/17205737d77ef7ea34cce64e95b1ffe0/9eccc64a4f976f9e37a62749b9204e694907d977

### Babel Configuration (.babelrc, package.json, cli command)
NA

### Expected Behavior
Based on the code in the link above, I expect the output to be:

```js
foo("cheese", "baz", function() {
  return "bop";
});
```

### Current Behavior

```js
foo(("cheese", "baz"), function() {
  return "bop";
});
```

### Possible Solution
NA

### Context
I'm trying to write a transform that automatically inserts dependent keys in to a computed property in an Ember codebase, based on some criteria.

### Your Environment
The bug is happening in the ASTExplorer as I'm spiking out my idea.

See ASTExplorer for version info

**Comment by loganfsmyth** [comment]
I'm estimating this as a relatively straightforward fix, but if someone takes it and finds otherwise, please report back.

This is _likely_ a fix that needs to be done in https://github.com/babel/babel/blob/3c4f19a28de34aafd842a475d6e7b2d4a0512d34/packages/babel-traverse/src/path/modification.js#L18 and we'll want to verify and possibly fix the behavior for `pushContainer` via this too: https://github.com/babel/babel/blob/3c4f19a28de34aafd842a475d6e7b2d4a0512d34/packages/babel-traverse/src/path/modification.js#L106

**Comment by achambers** [comment]
@loganfsmyth I'm keen to give this a crack. I found the test that I need to update so will start there and see how I get on. 

Feel free to adding this issue to me if that's possible. I'll put something together tomorrow.

**Comment by colshacol** [comment]
I have spent the last two hours trying to track this down. (Because I am bored, off work, and really want to begin contributing to Babel.)

I followed the data all through `babel-traverse/src/path` modification and replacement, but it does not seem to be happening there.

`packages/babel-traverse/src/path/replacement.js` finishes (in the particular case of `cheese`/`baz` on line `200`. And at that point, nothing has indicated to me that parens should be present anywhere in the forthcoming output.

On line `75` of the same file, I have found the `replacement = `(${replacement})`;` that seems like it would be the culprit, but this `replaceWithSourceString` function is actually never invoked during this test case.
